docs/src/main/asciidoc/config-extending-support.adoc

```diff
@@ -0,0 +1,338 @@
+
+[source,java]
+----
+package org.acme.config;
+
+import org.eclipse.microprofile.config.spi.Converter;
+
+public class MicroProfileCustomValueConverter implements Converter<MicroProfileCustomValue> {
+
+    @Override
+    public MicroProfileCustomValue convert(String value) {
+        return new MicroProfileCustomValue(Integer.parseInt(value));
+    }
+}
+----
+
+The custom configuration type converts the Config Source value automatically:
+
+[source,java]
+----
+@ConfigProperty(name = "configuration.value.name")
+MicroProfileCustomValue value;
+----
+
+=== Converter priority
+
+The `javax.annotation.Priority` annotation overrides the Converter priority and change converters precedence to fine
+tune converters execution order. By default, if no `@Priority` can be found on a converter, it's registered with a
+priority of `100` and all Quarkus core converters are registered with a priority of `200`. Consider:
+
+[source,java]
+----
+package org.acme.config;
+
+import javax.annotation.Priority;
+import org.eclipse.microprofile.config.spi.Converter;
+
+@Priority(150)
+public class MyCustomConverter implements Converter<MicroProfileCustomValue> {
+
+    @Override
+    public MicroProfileCustomValue convert(String value) {
+
+        final int secretNumber;
+        if (value.startsFrom("OBF:")) {
+            secretNumber = Integer.parseInt(SecretDecoder.decode(value));
+        } else {
+            secretNumber = Integer.parseInt(value);
+        }
+
+        return new MicroProfileCustomValue(secretNumber);
+    }
+}
+----
+
+Since it converts the same value type (namely `MicroProfileCustomValue`) and has a priority of `150`, it will be used
+instead of a `MicroProfileCustomValueConverter` which has a default priority of `100`.
+
+[[config-interceptors]]
+== Config Interceptors ==
+
+https://github.com/smallrye/smallrye-config[SmallRye Config] provides an interceptor chain that hooks into the
+Config value resolution. This is useful to implement features like Property Substitution, Configuration Profiles, or
+just Logging to find out where the config value was loaded from.
+
+An interceptor can be created by implementing the `io.smallrye.config.ConfigSourceInterceptor` interface. The
+`io.smallrye.config.ConfigSourceInterceptor` is able to intercept the resolution of a configuration name with the method
+`ConfigValue getValue(ConfigSourceInterceptorContext context, String name)`. The `ConfigSourceInterceptorContext` is
+used to proceed with the interceptor chain. The chain can be short-circuited by returning an instance of
+`io.smallrye.config.ConfigValue`. The `ConfigValue` objects hold information about the key name, value, config source
+origin and ordinal.
+
+NOTE: The Interceptor Chain is applied before any Conversion is performed on the configuration value.
+
+Registration of an interceptor in the interceptor chain is done via the `ServiceLoader` mechanism and provide the
```

Comment:
`and provide` is weird here. It should at least be `provides` but even that looks a bit weird. It's more `you should provide` than anything else.